File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        multiset<int> min_heap; //min heap implemetation
        for (int num : nums){
            min_heap.insert(num);
            if(min_heap.size() > k) min_heap.erase(min_heap.begin());

        }

        return *min_heap.begin();
    }
};

File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.py
class Solution(object):
    def findKthLargest(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        pq = []
        for val in nums:
            heapq.heappush(pq, val)
            if len(pq) > k:
                heapq.heappop(pq)

        return heapq.heappop(pq)